rust/src/lib.rs

```diff
@@ -847,12 +1493,75 @@ pub extern "C" fn zcashlc_get_received_memo_as_utf8(
     unwrap_exc_or_null(res)
 }
 
+/// Returns the memo for a received note by copying the corresponding bytes to the received
+/// pointer in `memo_bytes_ret`.
+///
+/// The note is identified by its row index in the `received_notes` table within the data
+/// database.
+///
+/// # Safety
+///
+/// - `db_data` must be non-null and valid for reads for `db_data_len` bytes, and it must have an
+///   alignment of `1`. Its contents must be a string representing a valid system path in the
+///   operating system's preferred representation.
+/// - The memory referenced by `db_data` must not be mutated for the duration of the function call.
+/// - The total size `db_data_len` must be no larger than `isize::MAX`. See the safety
+///   documentation of pointer::offset.
+/// - `memo_bytes_ret` must be non-null and must point to an allocated 512-byte region of memory.
+#[no_mangle]
+pub extern "C" fn zcashlc_get_received_memo(
+    db_data: *const u8,
+    db_data_len: usize,
+    id_note: i64,
+    memo_bytes_ret: *mut u8,
+    network_id: u32,
+) -> bool {
+    zcashlc_get_memo(
+        db_data,
+        db_data_len,
+        NoteId::ReceivedNoteId(id_note),
+        memo_bytes_ret,
+        network_id,
+    )
+}
+
+fn zcashlc_get_memo(
```

Comment:
Document safety requirements on this helper function (similar to other helper functions above).